pkg/apis/v1alpha1/serviceexport.go

```diff
@@ -66,13 +66,17 @@ const (
 	// service export has been recognized as valid by an mcs-controller.
 	// This will be false if the service is found to be unexportable
 	// (ExternalName, not found).
+	//
+	// Deprecated: use ServiceExportConditionValid instead
 	ServiceExportValid = "Valid"
 	// ServiceExportConflict means that there is a conflict between two
 	// exports for the same Service. When "True", the condition message
 	// should contain enough information to diagnose the conflict:
 	// field(s) under contention, which cluster won, and why.
 	// Users should not expect detailed per-cluster information in the
 	// conflict message.
+	//
+	// Deprecated: use ServiceExportConditionConflict instead
```

Comment:
I think the existing constant names are fine but if the consensus is to rename them then just change the existing constants instead of deprecating. I think deprecating is overkill in this case. It's not a big deal to modify a constant name after bumping the mcs-api version. Also, if one has linting that flags the use of deprecated fields then it would have to be changed anyway.

Reply:
To me the removal of those should probably be in a new version of the apis (v1alpha2 or v1beta1) to try to be mindful of people importing the package and using those. If you have linting that flags deprecated I would tend to say that it actually is achieving what's expected: aka you should probably change your conditions usage :thinking:.

Reply:
Well for instance there's a bunch of deprecation here related to conditions: https://github.com/kubernetes-sigs/gateway-api/blob/e1310bbd66c54b757d28ee65cf363825f6189ca5/apis/v1/gateway_types.go

My reasoning is that when project would use a newer version of the CRD they will import some other go package meaning that they will already need to actively change some code anyway and thus we would be able to not carry over certain deprecation such as this one.

For small version bump though I don't think we should make users of this library change their code by dropping those const IMO.